packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -197,11 +432,69 @@ export class Table extends Resource {
       actions: ['dynamodb:ListStreams'],
       resourceArns: ['*'],
     });
- }
+  }
+
+  /**
+   * Creates a Table construct that represents an external table.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs A `TableAttributes` object.
+   */
+  public static fromTableAttributes(scope: Construct, id: string, attrs: TableAttributes): ITable {
+
+    class Import extends TableBase {
+      public readonly tableName = this.parseTableName(scope, attrs);
+      public readonly tableArn = this.parseTableArn(scope, attrs);
+      public readonly tableStreamArn = attrs.tableStreamArn;
+
+      private parseTableName(construct: IConstruct, props: TableAttributes): string {
+        // if we have an explicit table name, use it.
+        if (props.tableName) {
+          return props.tableName;
+        }
+
+        // extract table name from table arn
+        if (props.tableArn) {
+          return Stack.of(construct).parseArn(props.tableArn).resourceName!;
+        }
+
+        // no table name is okay since it's optional.
+        throw new Error('Cannot determine table name. At least `tableArn` or `tableName` is needed');
+      }
+
+      private parseTableArn(construct: IConstruct, props: TableAttributes): string {
+
+        // if we have an explicit table ARN, use it.
+        if (props.tableArn) {
+          return props.tableArn;
+        }
+
+        if (props.tableName) {
+          return Stack.of(construct).formatArn({
+            // DynamoDB names are globally unique in a partition,
+            // and so their ARNs have empty region and account components
+            region: '',
+            account: '',
+            service: 'dynamodb',
+            resource: `table/${props.tableName}`
+          });
+        }
+
+        throw new Error('Cannot determine table ARN. At least `tableArn` or `tableName` is needed');
```

Comment:
> These private methods are not needed. This logic should be part of the body of the fromTableAttributes function.

For example, it will prevent the problem of doing this validation twice, while it's needed only once :).